crates/extrinsics/src/map_account.rs

```diff
@@ -144,18 +145,27 @@ where
     /// Returns the dry run simulation result, or an error in case of failure.
     pub async fn map_account_dry_run(&self) -> Result<()> {
         let call = MapAccount::new().build();
-        let res =
+        let bytes =
```

Comment:
New API/result type, including new `DryRunDecodeError` error which needs handling and bubbling up.